core/src/main/java/org/stellar/anchor/asset/JsonAssetService.java

```diff
@@ -19,12 +19,22 @@ public List<AssetInfo> listAllAssets() {
     return new ArrayList<>(assets.assets);
   }
 
-  public AssetInfo getAsset(String code, String issuer) {
+  public AssetInfo getAsset(String code) {
     for (AssetInfo asset : assets.assets) {
       if (asset.getCode().equals(code)) {
-        if (issuer == null || issuer.equals(asset.getIssuer())) {
           return asset;
-        }
+      }
+    }
+    return null;
+  }
+
+  public AssetInfo getAsset(String code, String issuer) {
+    if (issuer == null) {
+      return getAsset(code);
+    }
+    for (AssetInfo asset : assets.assets) {
+      if (asset.getCode().equals(code) && issuer.equals(asset.getIssuer())) {
```

Comment:
💡 nit: Can we use the safer version of comparison in case `asset.code()` is null? 
```java
if (code.equals(asset.getCode()) && issuer.equals(asset.getIssuer())) {
```